File: golang/35.search-insert-position.go
func searchInsert(nums []int, target int) int {
    left, right := 0, len(nums)
    var mid int

    for left < right {
        mid = (right + left) / 2

        if nums[mid] < target {
            left = mid + 1
        } else {
            right = mid
        }
    }

    return left
}

File: python/35.search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        def bs(lo, hi):
            if lo == hi:
                return lo

            mid = (lo + hi) // 2

            if nums[mid] < target:
                return bs(mid + 1, hi)
            else:
                return bs(lo, mid)
        
        return bs(0, len(nums))
